File: Modules/Payments/Http/Resources/PaymentMethodResource.php
<?php

namespace Modules\Payments\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;

class PaymentMethodResource extends JsonResource
{
    public function toArray($request): array
    {
        return [
            'id' => $this->id,
            'card_holder' => $this->card_type,
            'card_number' => $this->card_number,
            'expiry' => $this->expiry,
            'is_default' => $this->is_default,
        ];
    }
}

File: Solid.Api/Features/Payments/PaymentMethodResource.cs
using Solid.Api.Database.Entities;

namespace Solid.Api.Features.Payments;

public static class PaymentMethodResource
{
    public static object From(PaymentMethod method)
    {
        return new
        {
            id = method.Id,
            card_holder = method.CardHolder ?? method.CardType,
            card_number = method.CardNumber,
            expiry = method.Expiry,
            is_default = method.IsDefault
        };
    }
}
